### PR TITLE
f32: add SIMD Log, Log2, Log10, Pow, PowElem kernels (AVX2+FMA, NEON)

### LINKED ISSUE #109: f64/f32: add vectorized Log and Pow (Exp exists, Log/Pow missing)
## Summary

Add vectorized **`Log`** and **`Pow`** to `f64` (and `f32`). The library already has `Exp`, `Sigmoid`, `Tanh`, and `ReLU`, but no logarithm or power, which leaves a gap in the transcendental set for DSP and ML normalization.

## Motivation

`pow(x, p)` shows up directly in audio feature normalization. PCEN, for example, computes terms like `(S / (eps + M)^gain + bias)^power`. The natural vectorized form is `pow(x, p) = exp(p * log(x))`, but with only `Exp` available, the `log` has to be done with a scalar `math.Log` loop, which breaks vectorization right in the middle of an otherwise SIMD pipeline. Log-mel spectrograms and dB conversion (`10 * log10`) hit the same wall.

## Proposed API

```go
// natural log, plus the common bases
func Log(dst, src []float64)
func Log10(dst, src []float64)   // for dB / log-mel
func Log2(dst, src []float64)

// scalar exponent is the common DSP case (e.g. ^0.35 in PCEN)
func Pow(dst, src []float64, exp float64)

// optional: elementwise base/exp
func PowElem(dst, base, exp []float64)
```

Plus the `f32` equivalents, and in-place variants (`LogInPlace`, `PowInPlace`) to match the `Exp`/`Sigmoid` style.

## Accuracy and edge cases

- Match the accuracy/ULP approach already used by `Exp` (there are existing `TestExpAccuracy` style tests to mirror).
- Define and test edge behavior: `Log(0) = -Inf`, `Log(negative) = NaN`, `Pow` with `0` base / negative base / non-integer exponent, and saturation, the same way `Exp` handles overflow.
- `Pow(x, p)` can be implemented as `Exp(p * Log(x))` internally once `Log` exists, which keeps it on the fast path; a direct kernel can come later if profiling justifies it.

## Use cases

- PCEN and log-mel audio front-ends (pairs with the STFT request).
- dB / loudness conversion.
- General activation/normalization math where a power curve is needed.

## Acceptance criteria

- Bit-parity against a scalar Go reference (`*_Ref` pattern), allocation-free, concurrent-safe.
- ARM64 NEON + scalar fallback consistent with the rest of the library; a correct scalar fallback first is fine, with SIMD kernels gated on profiling.

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Vectorizes the f32 transcendental log/pow family from #119, completing #109 (the f64 half landed in #123).

- **Log core**: the same `x = m*2^e` integer reduction as the f64 kernels (offset `0x3f350000`, arithmetic shift 23), `m` in `[sqrt2/2, sqrt2)`, then the Cephes logf form `ln(m) = z - 0.5*z^2 + z^3*P(z)` with `z = m-1` and the degree-8 minimax polynomial. Unlike the f64 atanh form this needs no division. One parametrized kernel serves `Log`/`Log2`/`Log10` via hi/lo reconstruction constants. Measured max relative error **~1.4e-7 (about 1.2 float32 ulps)** over 3M random inputs including subnormals (pre-scaled by `2^32`). Specials blend to match `math.Log`.
- **Pow / PowElem**: fused `exp(p*ln(x))` through the existing degree-5 exp core, `y` clamped to `[-104, 89]` with the same split `2^k` reconstruction as the f64 kernels (#123), so overflow goes to `+Inf` and underflow degrades gradually through subnormals to 0, matching `math.Pow`'s classes. Accuracy ~1.4e-5 relative (log error amplified by `|y| <= 88`). Non-positive/non-finite bases and zero/non-finite exponents dispatch to the exact scalar path.
- The f32 scalar references need no subnormal fixes: they compute through float64 math, where float32 subnormals are normal values (unlike the f64 case fixed in #123).
- amd64: 8-lane AVX2+FMA; arm64: 4-lane hand-encoded NEON `WORD`s verified by asmcheck and cross-checked against `aarch64-linux-gnu-as`.

## Related Issues

Closes #109. Builds on #119 and #123.

## Benchmarks (1024 elements)

| Op | i5-11400F scalar | i5-11400F SIMD | Cortex-A76 scalar | Cortex-A76 NEON |
|----|------------------|----------------|--------------------|------------------|
| Log | 6.1 us | 0.56 us (10.8x) | 17.3 us | 3.6 us (4.8x) |
| Pow | 57.6 us | 1.7 us (34x) | 78.9 us | 9.2 us (8.5x) |

## Test Plan

- [x] Full suite + golangci-lint (0 issues) on amd64 (AVX2+FMA) and arm64 (rpi5, NEON), rebased onto post-#123 main
- [x] Tolerance-based parity (5e-7 log, 1e-4 pow), accuracy sweeps (full range, near-1, subnormals, sqrt(2) boundaries), edge-class pins, lengths 1-40, overflow classes
- [x] Differential fuzz targets `FuzzF32Log` / `FuzzF32Pow` actively fuzzed 40s on amd64 and 30s on arm64
- [x] asmcheck WORD/comment cross-validation

## Release Notes

* **Performance Improvements**
  * Accelerated logarithm and power transcendental operations (log, log2, log10, pow) using SIMD vectorization on AVX2+FMA and NEON-capable processors for improved computational speed.

* **Tests**
  * Added comprehensive test coverage for logarithm and power operations including edge cases, accuracy validation, and fuzz testing.

### CODE REVIEW

File: f32/f32_amd64.s [hunk 1]
```diff
@@ -6210,3 +6210,815 @@ euclid32_avx_sqrt:
...
+    MOVL AX, R10
+    SUBL BX, R10                       // R10 = tmp = bits - OFF
+    MOVL R10, R11
+    SARL $23, R11                      // R11 = e
+    ADDL R9, R11                       // e += bias
+    MOVL $0xFF800000, BX
+    ANDL BX, R10
+    SUBL R10, AX                       // AX = bits(m)
+    VMOVD AX, X2                       // X2 = m
+    CVTSL2SS R11, X3                   // X3 = e as float32
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_amd64.s]: Switched to the VEX three-operand form in 628b23b. Go's assembler spells it VCVTSI2SSL, not VCVTSL2SS. Also applied the same fix to the three CVTSQ2SD sites in the f64 kernels from #123 (VCVTSI2SDQ), since they had the identical issue. The older CVTSQ2SD uses elsewhere in f64_amd64.s are SSE2-only paths and keep the legacy encoding on purpose.

File: f32/f32_amd64.s [hunk 2]
```diff
@@ -6210,3 +6210,815 @@ euclid32_avx_sqrt:
...
+    MOVL AX, R10
+    SUBL BX, R10                       // tmp
+    MOVL R10, R11
+    SARL $23, R11                      // e
+    ADDL R9, R11
+    MOVL $0xFF800000, BX
+    ANDL BX, R10
+    SUBL R10, AX                       // bits(m)
+    VMOVD AX, X2                       // m
+    CVTSL2SS R11, X3                   // e
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_amd64.s]: Fixed in 628b23b, same as above.

File: f32/f32_amd64.s [hunk 3]
```diff
@@ -6210,3 +6210,815 @@ euclid32_avx_sqrt:
...
+    MOVL AX, R10
+    SUBL BX, R10                       // tmp
+    MOVL R10, R11
+    SARL $23, R11                      // e
+    ADDL R9, R11
+    MOVL $0xFF800000, BX
+    ANDL BX, R10
+    SUBL R10, AX                       // bits(m)
+    VMOVD AX, X2
+    CVTSL2SS R11, X3                   // e
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_amd64.s]: Fixed in 628b23b.

File: f32/f32_amd64.s [hunk 4]
```diff
@@ -6210,3 +6210,815 @@ euclid32_avx_sqrt:
...
+
+    VMULSS X14, X3, X5                 // e * k1lo
+    VFMADD231SS X13, X1, X5            // += lnm * k2
+    VFMADD231SS X15, X3, X5            // += e * k1hi
+    VMOVSS X5, (DX)
+    JMP  log32_scalar_next
+
+log32_scalar_nan:
+    MOVL $0x7FC00000, AX
+    MOVL AX, (DX)
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_amd64.s]: Done in the follow-up commit. This works for f32 because MOVL takes a 32-bit immediate store; the equivalent f64 blocks need the register round-trip since there is no 64-bit immediate store form.

File: f32/f32_amd64.s [hunk 5]
```diff
@@ -6210,3 +6210,815 @@ euclid32_avx_sqrt:
...
+    JMP  log32_scalar_next
+
+log32_scalar_neginf:
+    MOVL $0xFF800000, AX
+    MOVL AX, (DX)
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_amd64.s]: Done, same commit.

File: f32/f32_amd64.s [hunk 6]
```diff
@@ -6210,3 +6210,815 @@ euclid32_avx_sqrt:
...
+    JMP  log32_scalar_next
+
+log32_scalar_posinf:
+    MOVL $0x7F800000, AX
+    MOVL AX, (DX)
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_amd64.s]: Done, same commit.
